pkg/signatures/public_key_fetcher.go

```diff
@@ -52,6 +60,12 @@ func (c *cosignPublicKeySignatureFetcher) FetchSignatures(ctx context.Context, i
 		return nil, err
 	}
 
+	// Wait until the registry rate limiter allows entrance.
+	err = c.registryRateLimiter.Wait(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "waiting for rate limiter entrance")
```

Comment:
nit: can we add the registry name to the error either here or up in the execution chain?